utils/defs.h

```diff
@@ -143,10 +150,9 @@ struct config_options
     };
 
     /* Reserving to maintain ABI compatibility for the future */
-    /* Note that the math here assumes no padding bytes, that's because of the
-       order in which the fields are declared (largest to smallest alignments)  */
```

Comment:
Probably good to keep the comment about padding bytes in - unless that has changed and now we do have padding bytes

Reply:
I noticed some of the existing fields were out of order, e.g. `uint16_t max_cells_per_dim` comes after `uint8_t` fields. Likewise I think the `binning_flags` union may require 4-byte alignment but comes after the uint8s.

Since we're already changing the ABI, I'm happy to swap these around if we think that's better.